superbench/benchmarks/micro_benchmarks/dist_inference.py

```diff
@@ -0,0 +1,359 @@
+            self.__local_rank = int(os.environ['LOCAL_RANK'])
+        except BaseException as e:
+            self._result.set_return_code(ReturnCode.DISTRIBUTED_SETTING_INIT_FAILURE)
+            torch.distributed.destroy_process_group()
+            logger.error('Initialize distributed env failed - benchmark: {}, message: {}.'.format(self._name, str(e)))
+            return False
+
+        if torch.cuda.is_available():
+            torch.cuda.set_device(self.__local_rank)
+            self.__device = torch.device('cuda:{}'.format(self.__local_rank))
+            self.__cuda_available = True
+        else:
+            self.__device = torch.device('cpu:{}'.format(self.__local_rank))
+            self.__cuda_available = False
+
+        return True
+
+    def _benchmark(self):
+        """Implementation for benchmarking."""
+        batch_size = self._args.batch_size
+        input_size = self._args.input_size
+        hidden_size = self._args.hidden_size
+        num_layers = self._args.num_layers
+        computation = self._args.computation_kernel
+        communication = self._args.communication_kernel
+        activation = self._args.activation_kernel
+        precision = self._args.precision
+        num_warmup = self._args.num_warmup
+        num_steps = self._args.num_steps
+
+        if self.__local_rank == 0:
+            logger.info(
+                'Distributed Inference - using {} GPUs: '
+                'batch_size={}, input_size={}, hidden_size={}, num_layers={}, '
+                'computation_kernel={}, communication_kernel={}, activation_kernel={}, precision={}, '
+                'num_warmup={} num_steps={}'.format(
+                    self.__world_size, batch_size, input_size, hidden_size, num_layers, computation, communication,
+                    activation, precision, num_warmup, num_steps
+                )
+            )
+
+        model = DistInferenceModel(
+            input_size, hidden_size, num_layers, precision, computation, communication, activation, self.__device,
+            self.__world_size
+        )
+        data = torch.rand(batch_size, input_size, dtype=getattr(torch, precision.value), device=self.__device)
+
+        # warm up
+        warmup_step_times = []
+        for i in range(self._args.num_warmup):
+            start = self.__timer()
+            model.forward(data)
+            end = self.__timer()
+            warmup_step_times.append((end - start) * 1000)
+
+        # run and collect results
+        test_step_times = []
+        for i in range(self._args.num_steps):
+            start = self.__timer()
+            model.forward(data)
+            end = self.__timer()
+            test_step_times.append((end - start) * 1000)
+
+        if not self._process_numeric_result('warmup_step_times', warmup_step_times, cal_percentile=True):
+            return False
+        if not self._process_numeric_result('test_step_times', test_step_times, cal_percentile=True):
```

Comment:
We should have each layer metrics as well.

Reply:
Fixed